common/src/ethereum/walletconnect.rs

```diff
@@ -12,6 +12,14 @@ pub struct WalletConnect {
     address: Address,
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum WalletError {
+    #[error(transparent)]
+    CallError(#[from] CallError),
+    #[error("failed to sign the tx")]
+    FailedToSignTx,
```

Comment:
We usually don't include "error/failed" in the name of error variants, as it's redundant, so I'd suggest `WalletError::Call` and `WalletError::TransactionSignature`.

We should also improve the error message of the latter, as it'll be displayed to users, but I'm not sure in what case that would fail?

Reply:
Fixed.

I noticed sometimes the raw response from API was lacking the signature part. Perhaps it was due to how spinner interferes with stdout, because once I disabled spinner, I wasn't able to reproduce this.

But regardless, if we ever fail to extract v,r,s from sig, we'll show that error which is more informative.